torchgeo/datasets/flair2.py

```diff
@@ -0,0 +1,659 @@
+                tensor[-1] = torch.div(tensor[-1], 5)
+            
+        return tensor
+
+    def _load_sentinel(self, paths: dict[str, Path], aerial_path: Path) -> tuple[list[Tensor], tuple[slice, slice]]:
+        # FIXME: should this really be returned as a tuple?
+        """Load a sentinel array.
+
+        Args:
+            paths (list[Path]): paths to sentinel data and snow cloud mask
+            aerial_path (Path): path to the corresponding aerial image, required for loading from centroids file
+
+        Returns:
+            Sequence[Tensor]: ground truth and snow cloud mask as tensors of shape TxCxHxW (time, channels, height, width)
+        """
+        data = torch.from_numpy(np.load(paths["data"])).float()
+        snow_cloud_mask = torch.from_numpy(np.load(paths["snow_cloud_mask"])).float()
+        
+        img_id = os.path.basename(aerial_path)
+        centroid_x_y = cast(tuple[int, int], self.centroids[img_id])
+        img, cropping_indices = self._crop_super_patch(data, centroid_x_y)
+        masks, _ = self._crop_super_patch(snow_cloud_mask, centroid_x_y)
+        
+        return [img, masks], cropping_indices
+        
+    def _load_target(self, path: Path) -> Tensor:
+        """Load a single mask corresponding to image.
+
+        Args:
+            path: path to the mask
+
+        Returns:
+            Tensor: the mask of the image
+        """
+        with rasterio.open(path) as f:
+            array: np.typing.NDArray[np.int_] = f.read(1)
+            tensor = torch.from_numpy(array).long()
+            # According to datapaper, the dataset contains classes beyond 13
+            # however, those are grouped into a single "other" class
+            # Rescale the classes to be in the range [0, 12] by subtracting 1
+            torch.clamp(tensor - 1, 0, len(self.classes) - 1, out=tensor)
+            
+        return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset."""
+        # Check if centroids metadata file or zip is present
+        if not os.path.isfile(os.path.join(self.root, f"{self.centroids_file}.json")):
+            if not os.path.isfile(os.path.join(self.root, f"{self.centroids_file}.zip")):
+                if not self.download:
+                    raise DatasetNotFoundError(self)
+                self._download(self.centroids_file)
+            self._extract(self.centroids_file)
+        
+        # Files to be extracted
+        to_extract: list[str] = []
+
+        # Check if dataset files (by checking glob) are present already
+        for train_or_test, dir_name in self.dir_names[self.split].items(): 
+            downloaded_path = os.path.join(self.root, dir_name)
+            if not os.path.isdir(downloaded_path):
+                to_extract.append(dir_name)
+                continue
+            
+            files_glob = os.path.join(downloaded_path, "**", self.globs[train_or_test])
```

Comment:
When instantiating the datamodule, it will extract the `flair_sen_train` zip-file on every run. I traced it here, and I believe it's becuase when going through when `dir_name="flair_sen_train"`, I get `files_glob="/home/jhj/dataset_test/FLAIR2/flair_sen_train/**/SEN2_*{0}.npy"`, with the curly braces at the end. This makes the `glob.glob()` fail in the next line. I tried manually setting it to `files_glob="/home/jhj/dataset_test/FLAIR2/flair_sen_train/**/SEN2_*.npy"`, which fixed the issue.

Reply:
Seems like it's the same with `flair_2_sen_test`, which also gets the `{0}` in `files_glob`, and extracts the zip-file on every run.

Reply:
> When instantiating the datamodule, it will extract the `flair_sen_train` zip-file on every run. I traced it here, and I believe it's becuase when going through when `dir_name="flair_sen_train"`, I get `files_glob="/home/jhj/dataset_test/FLAIR2/flair_sen_train/**/SEN2_*{0}.npy"`, with the curly braces at the end. This makes the `glob.glob()` fail in the next line. I tried manually setting it to `files_glob="/home/jhj/dataset_test/FLAIR2/flair_sen_train/**/SEN2_*.npy"`, which fixed the issue.

I see why the unexpected behavior appears. For clarification: the format string is necessary, as there are two files inside the directory: `mask` and `data` files. To be able to respectively get only the corresponding mask or data file, I had to format the glob.

Would you be willing to share with me your code snippet so I can debug this real quick?

Reply:
I just manually removed the `{0}` in the `files_glob` string before running `glob.glob()`:
```
files_glob = os.path.join(downloaded_path, "**", self.globs[train_or_test])
if "flair_sen_train" in files_glob:
    files_glob="/home/jhj/dataset_test/FLAIR2/flair_sen_train/**/SEN2_*.npy"
if not glob.glob(files_glob, recursive=True):
    to_extract.append(dir_name)
```
I.e., hard-coded `files_glob` so I could check if it stopped extracting the zip-file on every run. Not really a solution, but just wanted to see if I was the right place in the code.